src/main/java/de/uhd/ifi/se/decision/management/jira/rest/impl/KnowledgeRestImpl.java

```diff
@@ -45,347 +47,400 @@
+		documentationLocationOfParentElement, user);
+	if (linkId == 0) {
+	    return Response.status(Status.INTERNAL_SERVER_ERROR)
+		    .entity(ImmutableMap.of("error", "Link could not be updated.")).build();
+	}
+	KnowledgeElement parentElement = persistenceManager
+		.getManagerForSingleLocation(documentationLocationOfParentElement)
+		.getDecisionKnowledgeElement(idOfParentElement);
+	persistenceManager.updateIssueStatus(parentElement, updatedElement, user);
+	return Response.status(Status.OK).build();
+    }
+
+    @Override
+    @Path("/deleteDecisionKnowledgeElement")
+    @DELETE
+    @Produces({ MediaType.APPLICATION_JSON })
+    public Response deleteDecisionKnowledgeElement(@Context HttpServletRequest request,
+	    KnowledgeElement decisionKnowledgeElement) {
+	if (decisionKnowledgeElement == null || request == null) {
+	    return Response.status(Status.BAD_REQUEST)
+		    .entity(ImmutableMap.of("error", "Deletion of decision knowledge element failed.")).build();
+	}
+	String projectKey = decisionKnowledgeElement.getProject().getProjectKey();
+	ApplicationUser user = AuthenticationManager.getUser(request);
 
-		boolean isDeleted = KnowledgePersistenceManager.getOrCreate(projectKey)
-				.deleteDecisionKnowledgeElement(decisionKnowledgeElement, user);
-		if (isDeleted) {
-			return Response.status(Status.OK).entity(true).build();
-		}
-		return Response.status(Status.INTERNAL_SERVER_ERROR)
-				.entity(ImmutableMap.of("error", "Deletion of decision knowledge element failed.")).build();
-	}
-
-	@Override
-	@Path("/createLink")
-	@POST
-	@Produces({ MediaType.APPLICATION_JSON })
-	public Response createLink(@Context HttpServletRequest request, @QueryParam("projectKey") String projectKey,
-			@QueryParam("knowledgeTypeOfChild") String knowledgeTypeOfChild, @QueryParam("idOfParent") long idOfParent,
-			@QueryParam("documentationLocationOfParent") String documentationLocationOfParent,
-			@QueryParam("idOfChild") long idOfChild,
-			@QueryParam("documentationLocationOfChild") String documentationLocationOfChild,
-			@QueryParam("linkTypeName") String linkTypeName) {
-		if (request == null || projectKey == null || idOfChild <= 0 || idOfParent <= 0) {
-			return Response.status(Status.BAD_REQUEST)
-					.entity(ImmutableMap.of("error", "Link could not be created due to a bad request.")).build();
+	boolean isDeleted = KnowledgePersistenceManager.getOrCreate(projectKey)
+		.deleteDecisionKnowledgeElement(decisionKnowledgeElement, user);
+	if (isDeleted) {
+	    return Response.status(Status.OK).entity(true).build();
+	}
+	return Response.status(Status.INTERNAL_SERVER_ERROR)
+		.entity(ImmutableMap.of("error", "Deletion of decision knowledge element failed.")).build();
+    }
+
+    @Override
+    @Path("/assignDecisionGroup")
+    @POST
+    @Produces({ MediaType.APPLICATION_JSON })
+    public Response assignDecisionGroup(@Context HttpServletRequest request, @QueryParam("sourceId") long sourceId,
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [The method 'assignDecisionGroup(HttpServletRequest, long, String, String, String, String, String)' has an NPath complexity of 224](https://app.codacy.com/manual/UHD/cures-condec-jira/pullRequest?prid=5160185)